### PR TITLE
TSGX-4 retrieve NFT details and store in enclave

### PR DESCRIPTION
**Pull request by tdelabro**
OK:
Event detection succesfully trigger NOOP ecall
Pass actual arguments to the enclave
Seal NFT data
Ecall to check if an accountID is the owner of a NFT

Postponed:
Check for transaction inclusion in the chain
Retrive blockId and transactionId

### CODE REVIEW

File: core-primitives/sgx/crypto/Cargo.toml [hunk 1]
```diff
@@ -8,22 +8,28 @@ edition = "2018"
...
 ofb = { version = "0.4.0" }
-codec  = { package = "parity-scale-codec", version = "2.0.0", default-features = false, features = ["derive"] }
+codec = { package = "parity-scale-codec", version = "2.0.0", default-features = false, features = [
+    "derive",
+] }
 derive_more = { version = "0.99.5" }
 log = { version = "0.4.14", default-features = false }
-serde = { version = "1.0", default-features = false, features = ["alloc"] , optional = true }
-serde_json = { version = "1.0", default-features = false, features = ["alloc"] , optional = true }
+serde = { version = "1.0", default-features = false, features = [
```
Comment (markopoloparadox) [inline core-primitives/sgx/crypto/Cargo.toml:16]: Let's keep the dependencies as one-liners :)

File: core-primitives/sgx/nfts/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		Self(id, data)
+	}
+}
+
+#[derive(Debug, Encode, Decode)]
+pub struct NftDb(Vec<Nft>);
+
+impl Default for NftDb {
+	fn default() -> Self {
+		Self([].to_vec())
```
Comment (markopoloparadox) [inline core-primitives/sgx/nfts/src/lib.rs:45]: Maybe vec![] or Vec::new()

File: core-primitives/sgx/crypto/Cargo.toml (hunk 1 above)
Comment (tdelabro) [reply core-primitives/sgx/crypto/Cargo.toml:16]: It's rust fmt auto formating. Cannot escape it, better get used to it :)

File: core-primitives/sgx/nfts/src/lib.rs (hunk 2 above)
Comment (tdelabro) [reply core-primitives/sgx/nfts/src/lib.rs]: done

**Pull request merged**